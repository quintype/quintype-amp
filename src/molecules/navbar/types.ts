import { Config } from "../../types/config";

export interface NavbarTypes {
  config: Config;
  logoSrc?: string;
  align?: "left" | "right";
}
