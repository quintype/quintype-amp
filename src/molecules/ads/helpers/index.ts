import get from "lodash.get";

export const getAdPropsFromConfig = ({ config, adName }) => get(config, ["ampConfig", "doubleclick", adName], null);
export const isEmpty = (obj: object) => Object.keys(obj).length === 0;
export const getDataSlotFromUnitPath = (adProps) => {
  const unitPath = get(adProps, "unit-path", null);
  if (unitPath) {
    delete adProps["unit-path"];
    adProps["data-slot"] = unitPath;
  }
  return adProps;
};

// export const getAdPropsFromConfig = ({ config, adName }) => get(config, ["ampConfig", "ads", adName], null);
//
// config = {  // in the future, this is how ads will be in the config
//   ampConfig: {
//     ...
//     ads: {
//       "top-ad": {
//         type: "industrybrains",
//         width: 300,
//         height: 250,
//         "data-width": "300",
//         "data-height": "250",
//         "data-cid": "19626-3798936394"
//       },
//       "bottom-ad": {
//         width: "320",
//         height: "50",
//         type: "doubleclick",
//         "data-slot": "/35096353/amptesting/formats/sticky"
//       }
//     },
//     ...
//   }
// }
