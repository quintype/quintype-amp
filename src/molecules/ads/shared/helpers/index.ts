import get from "lodash.get";
import cloneDeep from "lodash.clonedeep";

const isEmpty = (obj: object) => Object.keys(obj).length === 0;

export const getPropsForDfpAd = ({ overridingProps, config, adName }) => {
  if (isEmpty(overridingProps)) {
    const clone = cloneDeep(config);
    const adPropsFromConfig = get(clone, ["ampConfig", "doubleclick", adName], null);
    const unitPath = get(adPropsFromConfig, "unit-path", null);

    if (!adPropsFromConfig) throw new Error(`No props defined for ${adName} in config`);
    if (!unitPath) throw new Error(`unit-path not defined for ad ${adName} in config`);
    adPropsFromConfig["data-slot"] = unitPath; // renames unit-path to data-slot
    delete adPropsFromConfig["unit-path"];
    return adPropsFromConfig;
  } else return overridingProps;
};
