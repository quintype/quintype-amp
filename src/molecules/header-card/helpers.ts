import get from "lodash.get";
import { GetHeaderComponentHelperTypes } from "./types";

export const getHeaderComponentConfig = ({ componentName, config }: GetHeaderComponentHelperTypes) => {
  const components = get(config, ["opts", "headerCardConfig", "components"], []);
  const match = components.find((component) => component.name.toLowerCase() === componentName);
  if (!match)
    throw new Error(`Error in "getHeaderComponentConfig" function. Could not find "${componentName}" in config`);
  return match.config;
};
