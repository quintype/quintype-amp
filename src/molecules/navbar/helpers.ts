import { Config, MenuGroupItemsTypes } from "../../types/config";
import get from "lodash.get";

/**
 * returns hamburger menu items for subdomains
 * Will return "default" menu group items for main domain or if no menugroups are configured for subdomain
 *
 * which menuGroup to pick for which domain can be configured by setting featureConfig.sidebarMenu.menuGroupSlug for the subdomain
 * if featureConfig isn't passed, check if menuGroup with slug "amp-sidebar-menu-${domainSlug}" is present and return those items
 * if that's not set, check if menuGroup with slug "sidebar-menu-${domainSlug}" is present and return those items
 * if that's not set either, return default menu group items
 * return default menu group items in case there are no items in any of the above menuGroups
 *
 * @param {Object} config config object
 * @returns {Array} Menu Group Items that should be shown inside the hamburger menu
 */

export const getDomainSpecificHamburgerMenuItems = (config: Config) => {
  const { ampConfig } = config;
  if (!ampConfig["menu-groups"]) return [];
  const domainSlug = config.opts?.domainSlug;

  const defaultMenuGroupItems = get(config, ["ampConfig", "menu-groups", "default", "items"], []);
  if (!domainSlug) return defaultMenuGroupItems;

  const allMenuGroupsArr: MenuGroupItemsTypes[] = objToArr(ampConfig["menu-groups"]);

  if (allMenuGroupsArr.length) {
    const sidebarMenuGroupSlug = getSidebarMenuSlug(allMenuGroupsArr, config, domainSlug);
    if (sidebarMenuGroupSlug) {
      const menuGroupToReturn = allMenuGroupsArr.find((menuGroup) => menuGroup?.slug === sidebarMenuGroupSlug);
      if (menuGroupToReturn && menuGroupToReturn.items.length) return menuGroupToReturn.items;
    }
  }
  return defaultMenuGroupItems;
};

// "menu-groups" in api/v1/amp/config should've been an array but it's an obj. Therefore doing this hack
export const objToArr = (obj: object) => {
  const keysArr = Object.keys(obj);
  const arr: MenuGroupItemsTypes[] = [];
  keysArr.forEach((key) => obj[key] && arr.push(obj[key]));
  return arr;
};

const getSidebarMenuSlug = (arr, config, domainSlug) => {
  // returns slug of the menu group to take
  // returns menuGroupSlug from featureConfig OR "amp-sidebar-menu-<domainSlug>" OR "sidebar-menu-<domainSlug>" OR null in this order

  const menuGroupSlugFromFeatureConfig = get(config, ["opts", "featureConfig", "sidebarMenu", "menuGroupSlug"], null);
  const ampSidebarMenuPresent = arr.find((item) => item.slug === `amp-sidebar-menu-${domainSlug}`);
  const sidebarMenuPresent = arr.find((item) => item.slug === `sidebar-menu-${domainSlug}`);

  if (menuGroupSlugFromFeatureConfig) return menuGroupSlugFromFeatureConfig;
  else if (ampSidebarMenuPresent) return `amp-sidebar-menu-${domainSlug}`;
  else if (sidebarMenuPresent) return `sidebar-menu-${domainSlug}`;
  else return null;
};
