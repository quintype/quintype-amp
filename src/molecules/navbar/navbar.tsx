import React, { Fragment } from "react";
import styled from "styled-components";
import { PublisherLogoHeader, HamburgerMenu } from "../../atoms";
import { Hamburger } from "../../atoms/icons/hamburger";
import { NavbarTypes } from "./types";
import { withConfig } from "../../context";
import get from "lodash.get";
import { withTheme } from "styled-components";

const StyledNavbar = styled.header`
  width: 100%;
  height: 60px;
  position: relative;
`;
const LogoWrapperOuter = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;
const LogoWrapperInner = styled.div`
  position: relative;
  width: 200px;
  height: 50px;
`;
export const HamburgerWrapper = styled.div<{ align: "left" | "right" }>`
  position: absolute;
  top: 50%;
  cursor: pointer;
  ${(props) =>
    props.align === "left"
      ? `
    transform: translate(15px, -50%);
  `
      : `
    right: 0;
    transform: translate(-15px, -50%);
  `}
`;

export const NavbarBase = ({ align = "left", config, theme }: NavbarTypes) => {
  const isMenuEnabled = get(config, ["ampConfig", "menu", "enabled"], false);
  const hamburgerMenuItems = get(config, ["ampConfig", "menu-groups", "default", "items"], []);
  const textDirection = get(config, ["publisherConfig", "text-direction"], "ltr");
  const hamburgerColor = get(theme, ["color", "secondaryColor"], "currentColor");
  return (
    <StyledNavbar>
      <LogoWrapperOuter>
        <LogoWrapperInner>
          <PublisherLogoHeader />
        </LogoWrapperInner>
      </LogoWrapperOuter>
      {isMenuEnabled && hamburgerMenuItems.length > 0 && (
        <Fragment>
          <HamburgerWrapper role="button" tabIndex={0} on="tap:sidebar.open" align={align} aria-label="hamburger">
            <Hamburger width="40" height="40" color={hamburgerColor} />
          </HamburgerWrapper>
          <HamburgerMenu align={align} textDirection={textDirection} items={hamburgerMenuItems} />
        </Fragment>
      )}
    </StyledNavbar>
  );
};

export const Navbar = withTheme(withConfig(NavbarBase));
