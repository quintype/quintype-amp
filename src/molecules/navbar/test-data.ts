import { config } from "../../__fixtures__";
import cloneDeep from "lodash.clonedeep";

const domains = [
  {
    name: "astrology",
    slug: "astrology",
    "host-url": "https://astrology.raftaar.in",
    "beta-host-url": null,
    "home-collection-id": 41610,
    "section-ids": [21815],
    "menu-groups": []
  },
  {
    name: "News",
    slug: "news",
    "host-url": "https://news.raftaar.in",
    "beta-host-url": null,
    "home-collection-id": 36478,
    "section-ids": [18739],
    "menu-groups": [795, 797, 796]
  }
];

const dummyConfig1 = generateDummyConfig([
  "sidebar-menu-news",
  "default",
  "amp-sidebar-menu-news",
  "header-menu-health"
]);
dummyConfig1.publisherConfig.domains = [];
dummyConfig1.opts = { domainSlug: undefined };
// ----------------------------------------------------------------------

const dummyConfig2 = generateDummyConfig([
  "sidebar-menu-news",
  "default",
  "amp-sidebar-menu-news",
  "header-menu-health"
]);
dummyConfig2.publisherConfig.domains = domains;
dummyConfig2.opts = { domainSlug: null };
// ----------------------------------------------------------------------

const dummyConfig3 = generateDummyConfig([
  "sidebar-menu-news",
  "default",
  "amp-sidebar-menu-news",
  "header-menu-health"
]);
dummyConfig3.publisherConfig.domains = domains;
dummyConfig3.opts = { domainSlug: "astrology" };
// ----------------------------------------------------------------------

const dummyConfig4 = generateDummyConfig([
  "sidebar-menu-news",
  "default",
  "amp-sidebar-menu-news",
  "header-menu-health"
]);
dummyConfig4.publisherConfig.domains = domains;
dummyConfig4.opts = {
  domainSlug: "news",
  featureConfig: {
    sidebarMenu: {
      menuGroupSlug: "header-menu-health"
    }
  }
};
// ----------------------------------------------------------------------

const dummyConfig5 = generateDummyConfig([
  "sidebar-menu-news",
  "default",
  "amp-sidebar-menu-news",
  "header-menu-health"
]);
dummyConfig5.publisherConfig.domains = domains;
dummyConfig5.opts = { domainSlug: "news" };
// ----------------------------------------------------------------------

const dummyConfig6 = generateDummyConfig(["sidebar-menu-news", "default", "header-menu-health"]);
dummyConfig6.publisherConfig.domains = domains;
dummyConfig6.opts = { domainSlug: "news" };
// ----------------------------------------------------------------------

const dummyConfig7 = generateDummyConfig(["sidebar-menu-news", "default", "header-menu-health", "foo"]);
dummyConfig7.publisherConfig.domains = domains;
dummyConfig7.opts = {
  domainSlug: "news",
  featureConfig: {
    sidebarMenu: {
      menuGroupSlug: "foo"
    }
  }
};
// ----------------------------------------------------------------------

const dummyConfig8 = generateDummyConfig(["default"]);
dummyConfig8.ampConfig["menu-groups"]["amp-sidebar-menu-news"] = {
  id: 796,
  slug: "sidebar-menu-news",
  name: "Sidebar Menu News",
  items: []
};

dummyConfig8.publisherConfig.domains = domains;
dummyConfig8.opts = {
  domainSlug: "news"
};
// ----------------------------------------------------------------------

const dummyConfig9 = generateDummyConfig(["default"]);
dummyConfig9.ampConfig["menu-groups"]["sidebar-menu-news"] = {
  id: 796,
  slug: "sidebar-menu-news",
  name: "Sidebar Menu News",
  items: []
};

dummyConfig9.publisherConfig.domains = domains;
dummyConfig9.opts = {
  domainSlug: "news"
};
// ----------------------------------------------------------------------

function generateDummyConfig(slugsArr) {
  const dummyConfig = cloneDeep(config);
  const obj = {
    "sidebar-menu-news": {
      id: 796,
      slug: "sidebar-menu-news",
      name: "Sidebar Menu News",
      items: [
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 10821,
          title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 10821,
          "parent-id": null,
          url: "https://shabdkosh.raftaar.in",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
        },
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 10822,
          title: "\u092E\u094B\u092C\u093E\u0907\u0932 \u0917\u0947\u092E",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 10822,
          "parent-id": null,
          url: "https://www.raftaar.in/games",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://www.raftaar.in/games" }
        }
      ]
    },
    default: {
      id: 999,
      slug: "default",
      name: "Default",
      items: [
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 9874,
          title: "\u0916\u093C\u092C\u0930",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 9874,
          "parent-id": null,
          url: "https://news.raftaar.in",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://news.raftaar.in" }
        },
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 9875,
          title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 9875,
          "parent-id": null,
          url: "https://shabdkosh.raftaar.in",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
        },
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 9879,
          title: "\u092E\u094B\u092C\u093E\u0907\u0932 \u0917\u0947\u092E",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 9879,
          "parent-id": null,
          url: "https://www.raftaar.in/games",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://www.raftaar.in/games" }
        }
      ]
    },
    "amp-sidebar-menu-news": {
      id: 960,
      slug: "amp-sidebar-menu-news",
      name: "AMP Sidebar Menu News",
      items: [
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 12420,
          title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 12420,
          "parent-id": null,
          url: "https://shabdkosh.raftaar.in",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
        }
      ]
    },
    "header-menu-health": {
      id: 667,
      slug: "header-menu-health",
      name: "Header-Menu-Health",
      items: [
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": null,
          "entity-slug": null,
          "item-id": null,
          rank: 9598,
          title: "\u092C\u0940\u092E\u093E\u0930\u0940",
          "item-type": "link",
          "section-slug": null,
          "tag-slug": null,
          id: 9598,
          "parent-id": null,
          url: "https://health.raftaar.in/disease",
          "entity-name": null,
          "collection-slug": null,
          "section-name": null,
          data: { color: "#FFFFFF", link: "https://health.raftaar.in/disease" }
        },
        {
          "tag-name": null,
          "entity-properties": null,
          "collection-id": 36414,
          "entity-slug": null,
          "item-id": 18685,
          rank: 9599,
          title: "\u0928\u0941\u0938\u094D\u0916\u0947 \u0914\u0930 \u0926\u0947\u0916\u092D\u093E\u0932",
          "item-type": "section",
          "section-slug": "nuskhe-aur-dekhbhal",
          "tag-slug": null,
          id: 9599,
          "parent-id": null,
          url: "https://health.raftaar.in/nuskhe-aur-dekhbhal",
          "entity-name": null,
          "collection-slug": "nuskhe-aur-dekhbhal",
          "section-name": "\u0928\u0941\u0938\u094D\u0916\u0947 \u0914\u0930 \u0926\u0947\u0916\u092D\u093E\u0932",
          data: { color: "#FFFFFF", link: "https://health.raftaar.in/nuskhe-aur-dekhbhal" }
        }
      ]
    },
    foo: {
      id: 666,
      slug: "foo",
      name: "Foo",
      items: []
    }
  };
  const obj2 = {};
  slugsArr.forEach((slug) => (obj2[slug] = obj[slug]));
  dummyConfig.ampConfig["menu-groups"] = obj2;
  return dummyConfig;
}

const defaultMenuItems = [
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 9874,
    title: "\u0916\u093C\u092C\u0930",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 9874,
    "parent-id": null,
    url: "https://news.raftaar.in",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://news.raftaar.in" }
  },
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 9875,
    title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 9875,
    "parent-id": null,
    url: "https://shabdkosh.raftaar.in",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
  },
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 9879,
    title: "\u092E\u094B\u092C\u093E\u0907\u0932 \u0917\u0947\u092E",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 9879,
    "parent-id": null,
    url: "https://www.raftaar.in/games",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://www.raftaar.in/games" }
  }
];
const ampSidebarMenuNewsItems = [
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 12420,
    title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 12420,
    "parent-id": null,
    url: "https://shabdkosh.raftaar.in",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
  }
];
const sidebarMenuNewsItems = [
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 10821,
    title: "\u0936\u092C\u094D\u0926\u0915\u094B\u0936",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 10821,
    "parent-id": null,
    url: "https://shabdkosh.raftaar.in",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://shabdkosh.raftaar.in" }
  },
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 10822,
    title: "\u092E\u094B\u092C\u093E\u0907\u0932 \u0917\u0947\u092E",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 10822,
    "parent-id": null,
    url: "https://www.raftaar.in/games",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://www.raftaar.in/games" }
  }
];
const healthMenuItems = [
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": null,
    "entity-slug": null,
    "item-id": null,
    rank: 9598,
    title: "\u092C\u0940\u092E\u093E\u0930\u0940",
    "item-type": "link",
    "section-slug": null,
    "tag-slug": null,
    id: 9598,
    "parent-id": null,
    url: "https://health.raftaar.in/disease",
    "entity-name": null,
    "collection-slug": null,
    "section-name": null,
    data: { color: "#FFFFFF", link: "https://health.raftaar.in/disease" }
  },
  {
    "tag-name": null,
    "entity-properties": null,
    "collection-id": 36414,
    "entity-slug": null,
    "item-id": 18685,
    rank: 9599,
    title: "\u0928\u0941\u0938\u094D\u0916\u0947 \u0914\u0930 \u0926\u0947\u0916\u092D\u093E\u0932",
    "item-type": "section",
    "section-slug": "nuskhe-aur-dekhbhal",
    "tag-slug": null,
    id: 9599,
    "parent-id": null,
    url: "https://health.raftaar.in/nuskhe-aur-dekhbhal",
    "entity-name": null,
    "collection-slug": "nuskhe-aur-dekhbhal",
    "section-name": "\u0928\u0941\u0938\u094D\u0916\u0947 \u0914\u0930 \u0926\u0947\u0916\u092D\u093E\u0932",
    data: { color: "#FFFFFF", link: "https://health.raftaar.in/nuskhe-aur-dekhbhal" }
  }
];

export {
  defaultMenuItems,
  dummyConfig1,
  dummyConfig2,
  dummyConfig3,
  dummyConfig4,
  dummyConfig5,
  dummyConfig6,
  dummyConfig7,
  dummyConfig8,
  dummyConfig9,
  healthMenuItems,
  ampSidebarMenuNewsItems,
  sidebarMenuNewsItems
};
