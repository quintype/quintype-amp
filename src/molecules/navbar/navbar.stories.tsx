import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Navbar } from "./navbar";
import { config, textStory } from "../../__fixtures__";
import { Layout } from "../../atoms";
import cloneDeep from "lodash.clonedeep";

const modifiedConfig1 = cloneDeep(config);
modifiedConfig1.ampConfig.text_direction = "rtl";

const modifiedConfig2 = cloneDeep(config);
modifiedConfig2.ampConfig.menu.enabled = false;

storiesOf("Navbar", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <Navbar />)
  .add("With custom logo", () => <Navbar logoSrc="/header-logo.png" />)
  .add("Right Aligned", () => <Navbar align="right" />);

storiesOf("Navbar for RTL publisher", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={modifiedConfig1}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <Navbar />)
  .add("With custom logo", () => <Navbar logoSrc="/header-logo.png" />)
  .add("Right Aligned", () => <Navbar align="right" />);

storiesOf("Navbar for publisher with disabled hamburger menu", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={modifiedConfig2}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <Navbar />);
