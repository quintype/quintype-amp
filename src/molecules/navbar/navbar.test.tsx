import React from "react";
import { Navbar, HamburgerWrapper } from "./navbar";
import { mount, shallow } from "enzyme";
import { configWithMenuDisabled, configWithNoHamburgerMenuItems, configWithTextDirRtl } from "./navbar.stories";
import { Layout, HamburgerMenu } from "../../atoms";
import { textStory, config } from "../../__fixtures__";
import { Hamburger } from "../../atoms/icons/hamburger";
import { getDomainSpecificHamburgerMenuItems, objToArr } from "./helpers";
import {
  defaultMenuItems,
  dummyConfig1,
  dummyConfig2,
  dummyConfig3,
  dummyConfig4,
  dummyConfig5,
  dummyConfig6,
  dummyConfig7,
  dummyConfig8,
  dummyConfig9,
  dummyConfig10,
  healthMenuItems,
  ampSidebarMenuNewsItems,
  sidebarMenuNewsItems,
  sidebarMenuItemsMainDomain
} from "./test-data";

const LayoutWithMenuDisabled = () => (
  <Layout story={textStory} config={configWithMenuDisabled}>
    <Navbar />
  </Layout>
);
const LayoutWithNoHamburgerMenuItems = () => (
  <Layout story={textStory} config={configWithNoHamburgerMenuItems}>
    <Navbar />
  </Layout>
);
const LayoutWithRTL = () => (
  <Layout story={textStory} config={configWithTextDirRtl}>
    <Navbar />
  </Layout>
);
const DefaultLayoutWithNavbar = () => (
  <Layout story={textStory} config={config}>
    <Navbar />
  </Layout>
);
const LayoutWithNavbarAlignedRight = () => (
  <Layout story={textStory} config={config}>
    <Navbar align="right" />
  </Layout>
);

describe("Navbar", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = mount(<DefaultLayoutWithNavbar />);
    expect(wrapper.find(Navbar).exists()).toBeTruthy();
    wrapper.find("ul").forEach((node) => {
      expect(node.prop("dir")).toBe("ltr");
    });
  });
  it("Navbar with hamburger menu disabled", () => {
    const wrapper = mount(<LayoutWithMenuDisabled />);
    expect(wrapper.find(Hamburger).exists()).toBeFalsy();
    expect(wrapper.find(HamburgerMenu).exists()).toBeFalsy();
  });
  it("Navbar with no hamburger Menu Items", () => {
    const wrapper = mount(<LayoutWithNoHamburgerMenuItems />);
    expect(wrapper.find(Hamburger).exists()).toBeFalsy();
    expect(wrapper.find(HamburgerMenu).exists()).toBeFalsy();
  });
  it("Right aligned Navbar", () => {
    const wrapper = mount(<LayoutWithNavbarAlignedRight />);
    expect(wrapper.find(HamburgerWrapper).prop("align")).toBe("right");
  });
  it("Navbar with text direction RTL", () => {
    const wrapper = mount(<LayoutWithRTL />);
    wrapper.find("ul").forEach((node) => {
      expect(node.prop("dir")).toBe("rtl");
    });
  });
});

// tests for helper functions
describe("objToArr helper function", () => {
  it("should work", () => {
    const obj = {
      foo: {
        a: 1,
        b: "two"
      },
      bar: 42,
      baz: ["a", "b"]
    };
    expect(objToArr(obj)).toMatchObject([
      {
        a: 1,
        b: "two"
      },
      42,
      ["a", "b"]
    ]);
  });
});

describe("getDomainSpecificHamburgerMenuItems helper function", () => {
  it("Should return default menu group items for publisher having no associated subdomains", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig1);
    expect(menuItems).toMatchObject(defaultMenuItems);
  });
  it("Should return default menu group items when there are associated subdomains but the reader is on the main domain", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig2);
    expect(menuItems).toMatchObject(defaultMenuItems);
  });
  it("Should return default menu group items when the reader is on a subdomain but no featureConfig is passed and sidebar-menu-<subdomain> & amp-sidebar-menu-<subdomain> menuGroups are not created", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig3);
    expect(menuItems).toMatchObject(defaultMenuItems);
  });
  it("Should return menuGroup items as per featureConfig when the reader is on a subdomain, featureConfig is passed and sidebar-menu-<subdomain> & amp-sidebar-menu-<subdomain> menuGroups are created for the subdomain", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig4);
    expect(menuItems).toMatchObject(healthMenuItems);
  });
  it("Should return amp-sidebar-menu-<subdomain> menuGroup items when the reader is on a subdomain, featureConfig is not passed and sidebar-menu-<subdomain> & amp-sidebar-menu-<subdomain> menuGroups are created for the subdomain", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig5);
    expect(menuItems).toMatchObject(ampSidebarMenuNewsItems);
  });
  it("Should return sidebar-menu-<subdomain> menuGroup items when the reader is on a subdomain, featureConfig is not passed, amp-sidebar-menu-<subdomain> menuGroup isn't created and sidebar-menu-<subdomain> menuGroup is created for the subdomain", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig6);
    expect(menuItems).toMatchObject(sidebarMenuNewsItems);
  });
  it("Should return default menu group items when the reader is on a subdomain, and there are no menu items in menu groups amp-sidebar-menu-<subdomain>, sidebar-menu-<subdomain> or in the menuGroup given by featureConfig", () => {
    const menuItems1 = getDomainSpecificHamburgerMenuItems(dummyConfig7);
    const menuItems2 = getDomainSpecificHamburgerMenuItems(dummyConfig8);
    const menuItems3 = getDomainSpecificHamburgerMenuItems(dummyConfig9);
    expect(menuItems1).toMatchObject(defaultMenuItems);
    expect(menuItems2).toMatchObject(defaultMenuItems);
    expect(menuItems3).toMatchObject(defaultMenuItems);
  });
  it("Should return sidebar-menu menuGroup items when the reader is on main domain, featureConfig is not passed, sidebar-menu is created & amp-sidebar-menu menuGroup is not created", () => {
    const menuItems = getDomainSpecificHamburgerMenuItems(dummyConfig10);
    expect(menuItems).toMatchObject(sidebarMenuItemsMainDomain);
  });
});
