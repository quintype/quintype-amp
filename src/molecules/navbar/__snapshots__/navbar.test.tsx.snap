// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublisherLogoHeader should render 1`] = `
<Component
  config={Object {}}
/>
`;
